File: Brute_Force/javascript/1057.js
const [_, A, B] = require("fs").readFileSync("/dev/stdin").toString().split(" ");

const search = (a, b) => {
  let cnt = 1;
  while (Math.abs(a - b) !== 1 || Math.max(a, b) % 2) {
    cnt++;
    a = Math.ceil(a / 2);
    b = Math.ceil(b / 2);
  }
  return cnt;
};

const answer = search(A, B);
console.log(answer);

File: Brute_Force/python/1057.py
N,A,B = map(int,input().split())

def ceil(num):
  if num%1:
    return int(num)+1
  return int(num)

def search (a,b):
  round = 1
  while abs(a-b) != 1 or max(a,b) % 2:
    round += 1
    a = ceil(a/2)
    b = ceil(b/2)
  return round
print(search(A,B))